### DISCUSSION TITLE
Use of strongly typed Span for pattern indicators.

### DISCUSSION THREAD

**Discussion by markhammond** [hmG3/TA-Lib.NETCore#12] [General]
Hi @hmG3 !

Your modernisation changes look great, especially adopting Span. This enables (safe) incremental computation, something other C# TA-Lib ports lack.

I was intending to abstract over indicator outputs, and the introduction of strong(er) typing can make this difficult. Maybe I am over thinking this.
Are you sure you want to introduce a strongly typed Span<Core.CandlePatternType> for pattern indicators? 
One alternative would be to provide an interface to query the underlying type for each indicator output.

**Comment by hmG3** [comment]
Hi @markhammond. Thanks for reaching out.
Introduced `Span` definitely showed a significant performance boost in my benchmarking. I believe all modern implementations should leverage `Span` in function definitions these days. However, the next step to boosting performance even more would be introducing SIMD. I am going to do that after a stable release.

As for `Span<Core.CandlePatternType>` I also doubt it was a good idea :) The original concern was that candlestick functions and other functions share the same name for the output - `outInteger` and I intended to make a distinction between them.
This change was more like an experiment, and I will most likely revert it because it's better to describe those magic numbers in the documentation.

Currently, I am trying to establish the final contract for functions, release a new version, and then continue working on internal refactorings.
